File: 2P/Práctica4/C#/Practica4C#/Enemigo/LogicaEnemigo.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Practica4.Enemigo
{
    public class LogicaEnemigo
    {

        Random rand = new Random();
        Random rand2 = new Random();
        List<int> posiciones = new List<int>();
        
        public List<int> posicionBarcosEnemigo(int valorTamaño)
        {

            for (int i = 0; i < 5; i++)
            {
                int index = rand.Next(valorTamaño);
                
                if (!posiciones.Contains(index) || !(index < 1))
                {
                    posiciones.Add(index);
        
                }
                else
                {
                    index = rand.Next(valorTamaño);
                }
            }
          
            return posiciones;
        }


        public int eleccionEnemigo(int tamaño)
        {
            int aleatorio = rand2.Next(tamaño);
            int eleccion=0;

                if(!(aleatorio < 1))
                {
                    eleccion = aleatorio;
                }
                else
                {
                    aleatorio = rand2.Next(tamaño);
                }
            return eleccion;
        }
    }
}

File: 2P/Practica4/Java/src/Enemigo/LogicaEnemigo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Enemigo;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

/**
 *
 * @author Pablo Che
 */
public class LogicaEnemigo extends Thread{
    
    Random rand = new Random();
    Random rand2 = new Random();
    List <Integer> posiciones = new ArrayList<Integer>();
        
        public List<Integer> posicionBarcosEnemigo(int valorTamaño)
        {

            for (int i = 0; i < 5; i++)
            {
                int index = rand.nextInt(valorTamaño);
                
                if (!posiciones.contains(index) || !(index < 1))
                {
                    posiciones.add(index);
                }
                else
                {
                    index = rand.nextInt(valorTamaño);
                }
            }
          
            return posiciones;
        }

        public int eleccionEnemigo(int tamaño)
        {
            int aleatorio = rand2.nextInt(tamaño);
            int eleccion=0;

                if(!(aleatorio < 1))
                {
                    eleccion = aleatorio;
                }
                else
                {
                    aleatorio = rand2.nextInt(tamaño);
                }
            return eleccion;
        }
    
}
